cc/tf_bt_utils.cc

```diff
@@ -127,6 +128,46 @@ void WriteGameExamples(const std::string& gcp_project_name,
             << examples.size() << std::endl;
 }
 
+void WriteEvalRecord(const std::string& gcp_project_name,
+                     const std::string& instance_name,
+                     const std::string& table_name,
+                     const MctsPlayer& player,
+                     const std::string& black_player_name,
+                     const std::string& white_player_name) {
+  // TODO assert("eval" in table_name);
+  Table table(CreateDefaultDataClient(gcp_project_name, instance_name,
+                                      google::cloud::bigtable::ClientOptions()),
+                                      table_name);
+
+  // Retrieve the game counter from the Bigtable and increment it atomically.
+  using namespace google::cloud::bigtable;
+  auto rule = ReadModifyWriteRule::IncrementAmount("metadata", "game_counter", 1);
+  auto row = table.ReadModifyWriteRow("table_state", rule);
+  uint64_t game_counter = 0;
+  // TODO assert size == 1?
+  for (auto const& cell : row.cells()) {
+    if (cell.family_name() == "metadata" &&
+        cell.column_qualifier() == "game_counter") {
+      game_counter = cell.value_as<bigendian64_t>().get();
+      break;
+    }
+  }
+
+  auto row_name = absl::StrFormat(kEvalGameRowFormat, game_counter);
+  google::cloud::bigtable::SingleRowMutation row_mutation(
+      row_name,
+      SetCell("metadata", "black", black_player_name),
+      SetCell("metadata", "white", white_player_name),
+      SetCell("metadata", "black_won", player->result() > 0),
+      SetCell("metadata", "white_won", player->result() < 0)
+      SetCell("metadata", "result", player->result())
+      SetCell("metadata", "length", player->result() < 0)
+  );
```

Comment:
I'm not sure how to test this in production. advice would be appreciated.

Reply:
When I test, I build for a board size of 9 (for speed) and create a new table in the existing instance: `CBT_TABLE=games-9x9` (or any table that isn't the main one).

You may want to coerce/format your values to string as you pass them to `SetCell`. Even for integers, you may want them stored as strings so that they are more easily inspected. Use cases for storing an integer *as* an integer include `metadata:game_counter`, where you need to do an atomic set-and-get. These end up being big-endian integers, and are hard to read when viewing with `cbt`.

Reply:
Thanks for that knowledge :)